client/src/components/atoms/ImagePicker.tsx

```diff
@@ -1,24 +1,167 @@
+            validFile = false;
+        }
+        const fileReader = new FileReader();
+        fileReader.addEventListener(
+            "load",
+            async () => {
+                const res = fileReader.result;
+                console.log(res);
+                let imgStr;
+                if (res === null) imgStr = "";
+                else if (typeof res === "string") {
+                    imgStr = res;
+                } else {
+                    imgStr = res.toString();
+                }
+                const dimensions = await getImageDimensions(imgStr);
+                console.log(dimensions);
+                if (dimensions.width < 600 || dimensions.height < 430) {
+                    triggerWarning("Images must be at least 600 x 430 pixels");
+                    validFile = false;
+                }
+                if (validFile) {
+                    setShowWarning(false);
+                    onUploadImg(imgStr);
+                }
+            },
+            false
+        );
+        fileReader.readAsDataURL(file);
+    };
+
+    const clearImage = () => {
+        onUploadImg("");
+        onImageChange("");
+    };
+
+    const changeZoom = (increment: boolean) => {
+        if (increment) {
+            if (zoom >= 10) return;
+            setZoom(zoom + 0.1);
+        } else {
+            if (zoom <= 1) return;
+            setZoom(zoom - 0.1);
+        }
+    };
+    const triggerWarning = (text: string) => {
+        setShowWarning(true);
+        setWarningText(text);
+        setTimeout(() => {
+            setShowWarning(false);
+        }, 4000);
+    };
+    const imgInView = selected !== "" || uploadedImg !== "";
     return (
         <div className="imagepicker">
+            <WarningBox text={warningText} showWarning={showWarning} />
             <div className={`imagepicker-preview`}>
+                <button
+                    className={imgInView ? "btn btn-light imagepicker-trash" : "imagepicker-hidden"}
+                    onClick={clearImage}
+                >
+                    <i className="bi bi-trash"></i>
+                </button>
                 {selected.length ? (
                     <img src={selected} />
                 ) : (
-                    <div className={isErroneous ? "error" : "unselected"}>Select an image</div>
+                    <div className={isErroneous ? "error" : "unselected"} id="cropField">
+                        {uploadedImg !== "" ? (
+                            <Cropper
+                                image={uploadedImg}
+                                crop={crop}
+                                zoom={zoom}
+                                onCropChange={setCrop}
+                                onZoomChange={setZoom}
+                                maxZoom={10}
+                                cropSize={{
+                                    height: document.getElementById("cropField")!.offsetHeight,
```

Comment:
nit: useRef instead of DOM selectors (unsure of whether there are any consequences of React re-rendering when using document.getElementById)

Reply:
Updated.